wagtail_live/slack/receivers.py

```diff
@@ -0,0 +1,181 @@
+""" How to use it?
+
+To create an app, register your URL to slack, get a token plus all additional infrastucture, see:
+https://github.com/slackapi/bolt-python/tree/main/examples/django
+
+Then do something like this in your url_patterns:
+
+from wagtail_live.slack.views import slack_events_handler
+
+url_patterns += [
+    path("slack/events", slack_events_handler, name="slack_events_handler"),
+]
+
+Add the following to your settings.py file:
+
+LIVE_PAGE_MODEL = "model name"
+LIVE_APP = "app name"
+SLACK_BOT_TOKEN = "your slack bot token"
+SLACK_SIGNING_SECRET = "your slack signing secret"
+
+And you are good to go! (Soon hopefully!)
+
+"""
+
+import re
+
+import requests
+from django.apps import apps
+from django.conf import settings
+from django.core.files.base import ContentFile
+from django.utils.text import slugify
+from django.utils.timezone import now
+from wagtail.embeds.oembed_providers import all_providers
+from wagtail.images import get_image_model
+
+from ..blocks import (
+    construct_embed_block,
+    construct_image_block,
+    construct_live_post_block,
+    construct_text_block,
+)
+
+TEXT = "message"
+IMAGE = "image"
+EMBED = "embed"
+LivePost = "live_post"
+
+
+def is_embed(text):
+    for provider in all_providers:
+        for url_pattern in provider.get("urls", []):
+            # Somehow Slack links start with `<` and end with `>`.
+            if bool(re.match(url_pattern, text)):
+                return True
+
+    return False
+
+
+class SlackEventsAPIReceiver:
```

Comment:
This seems to have a lot of identical code to BaseMessageReceiver. Shouldn't it inherit from it to reduce duplicate code?

Reply:
Indeed, it should inherit from BaseMessageReceiver.